### ISSUE TITLE
consider adding piazza/slack/reddit integration to "report an issue"

### ISSUE THREAD

**Issue by tbretl** [PrairieLearn/PrairieLearn#1337]
Students have started using "report an issue" to ask about questions (e.g., "the correct answer does not make sense," "my code worked on another instance of this question but not on this one," etc.).

This is helpful in the sense that it exposes all the question data to the instructor and makes it easy to see what the student is asking about.

However, this breaks the standard policy in many courses of keeping all correspondence through a discussion forum (e.g., piazza, slack, reddit) - the only obvious way to respond is by email.

We might consider adding discussion forum integration, for example (1) auto-posting issues to forums, (2) adding a "share" button that creates a permalink to copy/paste into forum posts, etc.

**Comment by trombonekenny** [comment]
The how-to-engage-back-with-students-after-issues-are-reported does need improvement.

If there's a standard way to pass them somewhere else when they're reported, I think we should support that. That's nice because it still has all the question and student data, but you're entirely pushing tracking and responding to a different source. (We could optionally close the issue immediately on pass through so you don't see them in PL unless you go hunting for it.)

What would those options look like? Email it somewhere? Create a private post in piazza or slack, etc.? I'm not sure what's desired or technically available. 

The "share" button is trickier because you're pointing outsiders into PrairieLearn content where they might not have access. (Or maybe you want to respond but you don't want to share info about the question.) What do you propose that permalink points to?

**Comment by trombonekenny** [comment]
One solution might be a new app (PrairieIssues? PrairieTracker?) that acts like a traditional ticketing system but gets content and users from PrairieLearn and PrairieSchedule. It's independent enough that you can manage issues and incidents in that tracker app (to share them, have dialogue and status on them, etc.) while still pulling users and permissions from PL and PS. Different roles might get different views and access.

We'd use something like to digitize the CBTF incident reports. We just hadn't built that part yet.

**Comment by echuber2** [comment]
I'll add that it looks like there is no way to reply to an actual issue that is just a technical issue or to re-open issues, etc.

With reply threads on issues, maybe a workflow could look like:
1. Student submits issue on tracker
2. Staff get email notification about it (which they don't seem to currently?)
3. Staff can post something relevant on Piazza (etc.) about the resolution.
4. Staff can reply to the actual issue with a link to the Piazza post, then close and lock replies on the issue.

**Comment by nwalters512** [comment]
@jonatanschroeder is there anything not captured in your meta-issue? AFAICT the main points here (two-way communication and notifications) are already there.

**Comment by mwest1066** [comment]
Happy to close this.

**Comment by jonatanschroeder** [comment]
> @jonatanschroeder is there anything not captured in your meta-issue? AFAICT the main points here (two-way communication and notifications) are already there.

There is the discussion by @trombonekenny regarding a separate ticketing issue (https://github.com/PrairieLearn/PrairieLearn/issues/1337#issuecomment-430298571). Is that worth keeping?

**Comment by nwalters512** [comment]
No; we're not going to build a separate one. We may integrate incident reporting into PT in the longer run.
